py/makemoduledefs.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+# This pre-processor parses provided objects' c files for 
+# `MP_REGISTER_MODULE(module_name, obj_module, enabled_define)`
+# These are used to generate a header with the required entries for 
+# `mp_rom_map_elem_t mp_builtin_module_table[]` in py/objmodule.c
+
+from __future__ import print_function
+
+import re
+import os
+import argparse
+
+
+def find_c_file(vpath, obj_file):
+    """
+    Search the provided vpaths for the c file that matches the provided object_file
+    :param List[str} vpath: List of base paths, similar to gcc vpath
+    :param obj_file: object file to find the matching c file for
+    :return: path to c file or None
+    """
+    c_file = None
+    relative_c_file = os.path.splitext(obj_file)[0] + ".c"
+    relative_c_file = relative_c_file.lstrip('/\\')
```

Comment:
Since relative_c_file  this is passed to os.path.join this should not be needed I think?

Reply:
This is definitely needed, as the files being passed in from makefile with the build folder stripped off come in with a "/" at the start. 

Passing this in as the right hand arg of `os.path.join`, it gets treated as an absolute path and the left hand arg (from vpath) is ignored.

Reply:
>  it gets treated as an absolute path 

Well, that's new to me. Probably because the implementation is inconsistent between windows/linux and the type of slash used :(